disnake/message.py

```diff
@@ -1769,7 +1776,10 @@ async def create_thread(
         if self.guild is None:
             raise InvalidArgument("This message does not have guild info attached.")
 
-        default_auto_archive_duration: ThreadArchiveDuration = getattr(
+        if isinstance(auto_archive_duration, ThreadArchiveDuration):
+            auto_archive_duration = auto_archive_duration.value
+
+        default_auto_archive_duration: ThreadArchiveDurationT = getattr(
```

Comment:
```suggestion
        default_auto_archive_duration: ThreadArchiveDuration = getattr(
```
to be changed back if `channel.py:205` is changed as well